### PR TITLE
Implement search feature

### PR DESCRIPTION
**Pull request by paulacarneiro**
- Easily filter tags and notes by searching for a particular string

### CODE REVIEW

File: lib/store.ts [hunk 1]
```diff
@@ -166,6 +172,12 @@ export const useStore = create<UseStore>()((set) => ({
         },
       }));
     },
+    filteredNotes: [],
+    setFilteredNotes: (filteredNotes: NoteType[]) =>
```
Comment (davelange) [inline lib/store.ts:176]: This works, but I don't think it's the best approach for filtering. 
Filtered lists are basically state derived from other state, and in this case the search term applies to folders, notes and tags, so setting each thing separately with all the logic in the search bar isn't great. 

An option you could try out is having `searchTerm` on the main store, and then creating a custom hook (`useFilteredStore`) or something, that calculates the filtered lists based on the search term.

Like this you could also remove all the logic from the search bar component

File: components/search-bar.tsx [hunk 2]
```diff
@@ -0,0 +1,28 @@
+import { createRef, ChangeEvent } from "react";
+import { useStore } from "../lib/store";
+
+export default function SearchBar() {
+  const {
+    search: { setSearchTerm, searchTerm },
+  } = useStore();
+  const ref = createRef<HTMLInputElement>();
```
Comment (davelange) [inline components/search-bar.tsx:8]: This isn´t used anywhere

File: components/search-bar.tsx [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+
+  const reset = () => setSearchTerm("");
+
+  return (
+    <form onReset={reset} className="flex bg-white w-full justify-between">
+      <input
+        ref={ref}
+        className="bg-transparent w-full"
+        onChange={handleChange}
+        value={searchTerm}
```
Comment (davelange) [inline components/search-bar.tsx:22]: I don't think you need to set the `value` explicitly here

**Review by davelange — APPROVED** [review]

File: components/search-bar.tsx (hunk 3 above)
Comment (paulacarneiro) [reply components/search-bar.tsx:22]: Is there another way to do it?

Comment (davelange) [reply components/search-bar.tsx:28]: If you just remove it and leave the `onChange` it should work fine

**Review by davelange — APPROVED** [review]

**Pull request merged**